pkg/condition/condition.go

```diff
@@ -0,0 +1,129 @@
+package condition
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/google/cel-go/cel"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/openfga/openfga/pkg/condition/types"
+	"golang.org/x/exp/maps"
+)
+
+var emptyEvaluationResult = EvaluationResult{}
+
+type EvaluationResult struct {
+	ConditionMet bool
+}
+
+type EvaluableCondition struct {
```

Comment:
Would you please add a godoc comment here with some of the semantic details of the lazy loading of the condition, and also note that it is not safe for concurrent use at this time.